joplin/api/schema.py

```diff
@@ -6,19 +6,116 @@
+                rich_text_names.append(block.name)
+            elif block.child_blocks:
+                for child_block in block.all_blocks():
+                    if isinstance(child_block, RichTextBlock):
+                        rich_text_names.append(child_block.name)
+        except AttributeError as e:
+            pass
+    return rich_text_names
 
 
 class StreamFieldType(Scalar):
+    """
+    todo
+            for item in serialized:
+                loop through all the values and try running expand_db_html on them to return an href
+                still need to find a way to actually attach the janis_url to the selection tho..
+    """
+
     @staticmethod
-    def serialize(dt):
-        return [{'type': item.block_type, 'value': item.block.get_api_representation(item.value), 'id': item.id} for item in dt]
+    def serialize(StreamValue):
+        """
+        This is a rats nest of loops and conditionals, but the general idea is two functions:
+            1)loop through all the objects in the StreamValue, and
+                if it is a RichTextBlock,
+                add its name to a list of keys
+                ( this dosen't quite capture all the use cases as needed so it needs work)
+            2) use each of the above names as keys to sort through the stream_data
+                (which is a python dict represention of the StreamField, alot like JSON)
+                use the nested_lookup to recursivley return a list of all values that match the key
+                run a nested_alter using expand_db_html as a callback to edit those values in stream_data
+                (otherwise the values tend to get overwritten)
+
+        """
+        # get lists of potential keys
+        try:
+            rich_text_names = find_rich_text_names(StreamValue)
+        except Exception as e:
+            print(e)
+            print(traceback.format_exc())
+            pass
+
+            # todo: make things less nested
+        for block_key in rich_text_names:
+            if not isinstance(StreamValue, str):
+                data = StreamValue.stream_data
+            else:
+                data = ''
+            # sometimes data is just a string, and there isno stream_data
+            if len(data) is not 0:
+                # get all the vales that match the key
+                values = nested_lookup(block_key, StreamValue.stream_data)
+                for value in values:
+                    """
+                    if the value isnt a string, there is another nested list
+                    we need to alter each element of that list on its own to
+                    avoid accidentially overriding values
+                    """
+                    if not isinstance(value, str):
+                        for elem in value:
+                            for key in rich_text_names:
+                                print('alter elem', key)
+                                altered_value = nested_alter(elem, key, expand_db_html)
```

Comment:
or just the inner part of the loop, up to you

Reply:
I'm not sure what you mean

Reply:
```
for value in values:
   do_a_function_thing(value)